googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt

```diff
@@ -140,6 +151,59 @@ internal class DefaultGooglePayDelegate(
         AutoResolveHelper.resolveTask(paymentsClient.loadPaymentData(paymentDataRequest), activity, requestCode)
     }
 
+    override fun startGooglePayScreen(paymentDataLauncher: ActivityResultLauncher<Task<PaymentData>>) {
+        Logger.d(TAG, "startGooglePayScreen")
+        val paymentsClient = Wallet.getPaymentsClient(application, GooglePayUtils.createWalletOptions(componentParams))
+        val paymentDataRequest = GooglePayUtils.createPaymentDataRequest(componentParams)
+        val paymentDataTask = paymentsClient.loadPaymentData(paymentDataRequest)
+        coroutineScope.launch {
+            // no need to handle the result of this operation, paymentDataLauncher does that internally
+            runCatching { paymentDataTask.await() }
```

Comment:
We've put together a [brief extension function](https://github.com/google-pay/android-quickstart/blob/coroutines/kotlin/app/src/main/java/com/google/android/gms/samples/pay/viewmodel/CheckoutViewModel.kt#L152) that resolves the task through a continuation without processing its result. This can be used instead of `runCatching` and avoid polluting cancelations. See [example usage](https://github.com/google-pay/android-quickstart/blob/coroutines/kotlin/app/src/main/java/com/google/android/gms/samples/pay/activity/CheckoutActivity.kt#L71).

Reply:
Thanks! I just copied that code and pushed a commit, let me know your thoughts.

Reply:
LGTM